pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -73,6 +75,54 @@ impl<ForeignAsset> ForeignAssetCreatedHook<ForeignAsset> for () {
 	fn on_asset_created(_foreign_asset: &ForeignAsset, _asset_id: &AssetId) {}
 }
 
+/// Ensure origin location is a sibling
+fn convert_location<T>(location: &Location) -> Result<T::AccountId, DispatchError>
+where
+	T: Config,
+{
+	match location.unpack() {
+		(1, [Parachain(_)]) => T::ConvertLocation::convert_location(location)
+			.ok_or(Error::<T>::CannotConvertLocationToAccount.into()),
+		_ => Err(DispatchError::BadOrigin.into()),
+	}
+}
+#[derive(Decode, Encode, Debug, PartialEq, TypeInfo, Clone)]
+pub enum OriginType {
+	XCM(Location),
+	Governance,
+}
+
+/// Used to convert the success of an EnsureOrigin into `OriginType::Governance`
+pub struct MapSuccessToGovernance<Original>(PhantomData<Original>);
+impl<O, Original: EnsureOrigin<O, Success = ()>> EnsureOrigin<O>
+	for MapSuccessToGovernance<Original>
+{
+	type Success = OriginType;
+	fn try_origin(o: O) -> Result<OriginType, O> {
+		Original::try_origin(o)?;
+		Ok(OriginType::Governance)
+	}
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_successful_origin() -> Result<O, ()> {
+		Original::try_successful_origin()
+	}
+}
+
+/// Used to convert the success of an EnsureOrigin into `OriginType::XCM`
+pub struct MapSuccessToXcm<Original>(PhantomData<Original>);
+impl<O, Original: EnsureOrigin<O, Success = Location>> EnsureOrigin<O>
```

Comment:
Same as above, here it is a assumed that having a `Location` as `Success` type means that the Origin comes from XCM. In this case I would say is better than the other.